### PR TITLE
chore: always obsolete wal to avoid discontinuous log entries

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This changes the behavior during datanode startup in that it will always obsolete wal log entries to `flushed_entry_id` in manifest so that if it can tolerate inserts when the entry id span in logstore < `flushed_entry_id`.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Review by waynexia — APPROVED** [review]

File: src/mito2/src/region/opener.rs [hunk 1]
```diff
@@ -476,11 +474,8 @@ pub(crate) async fn replay_memtable<S: LogStore>(
         region_write_ctx.write_memtable();
     }
 
-    if allow_stale_entries && stale_entry_found {
-        wal.obsolete(region_id, flushed_entry_id, wal_options)
-            .await?;
-        info!("Force obsolete WAL entries, region id: {}, flushed entry id: {}, last entry id read: {}", region_id, flushed_entry_id, last_entry_id);
-    }
+    wal.obsolete(region_id, flushed_entry_id, wal_options)
```
Comment (evenyag) [inline src/mito2/src/region/opener.rs:477]: Maybe we can only do this when the last_id of the log store is not equal to `flushed_entry_id`.

**Review by evenyag — APPROVED** [review]
Mostly LGTM

File: src/mito2/src/region/opener.rs (hunk 1 above)
Comment (v0y4g3r) [reply src/mito2/src/region/opener.rs:477]: > Maybe we can only do this when the last_id of the log store is not equal to `flushed_entry_id`.

`LogStore` does not have a `span` API that returns current min/max entry id (only raft-engine provide that functionality).

Comment (killme2008) [reply src/mito2/src/region/opener.rs:477]: I think we should keep the info log.

```rust
info!("Force obsolete WAL entries, region id: {}, flushed entry id: {}, last entry id read: {}", region_id, flushed_entry_id, last_entry_id);
```

**Pull request merged**